File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        
//         int i = 0;
//         while (i != haystack.length() && needle[i] == haystack[i]) {
            
//             int j = i, k = i;
            
//             while (j != needle.length() && needle[j] == haystack[j]) {
                
//                 j++;
//             }
            
//             if (j == needle.length()) {
//                 return k;
//             }
            
//             i++;
//         }
        
        return haystack.indexOf(needle);
        
    }
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.scala
object Solution {
    def strStr(haystack: String, needle: String): Int = {


        // simpler
        haystack.indexOf(needle)

        // functional way
        // val n = haystack.length
        // val m = needle.length

        // (0 to n - m).find { i =>
        //     haystack.substring(i, i + m) == needle
        // }.getOrElse(-1)
    }
}